lightning/src/ln/peers/chacha.rs

```diff
@@ -3,21 +3,18 @@ use util::chacha20poly1305rfc::ChaCha20Poly1305RFC;
 
 pub const TAG_SIZE: usize = 16;
 
-pub fn encrypt(key: &[u8], nonce: u64, associated_data: &[u8], plaintext: &[u8]) -> Vec<u8> {
+pub fn encrypt(key: &[u8], nonce: u64, associated_data: &[u8], plaintext: &[u8], ciphertext_out: &mut [u8]) {
```

Comment:
It would be good to run some benchmarks tests with the difference. In principle, having a function mutate an input rather than returning the result is quite unidiomatic

Reply:
Here is how I think about it.

The original PR had design feedback that introducing memory allocations during the handshake was a red flag. So, part of my challenge with this large of a review is minimizing the number of behavior changes so that the restructure and testing work can go in and improve the codebase.

The code on master doesn't allocate for the handshake encryption, so I followed the same pattern to avoid excessive back-and-forths over preexisting design feedback.

I think if there are questions about memory allocation performance in the future those are easily addressed as separate PRs that focus on the behavior change in isolation.

I'm also not sure what benchmark tests you are suggesting. This was never performance feedback and was tied to memory allocation fragmentation which I don't believe there are easy tests to run.